src/main/java/ru/yandex/practicum/filmorate/controller/BasicController.java

```diff
@@ -0,0 +1,54 @@
+package ru.yandex.practicum.filmorate.controller;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import ru.yandex.practicum.filmorate.controller.validation.ValidationException;
+import ru.yandex.practicum.filmorate.model.Entity;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+public abstract class BasicController<T extends Entity> {
+    private int id;
+    private final String dtoName;
+    private final Map<Integer, T> idMapDTO = new HashMap<>();
+
+    protected BasicController(@NonNull String dtoName) {
+        this.dtoName = dtoName;
+    }
+
+    protected static Logger getLog() {
+        return log;
+    }
+
+    protected T addNew( T item) {
+        log.info(String.format("requested %s to add", dtoName));
```

Comment:
При логировании параметров, лучше использовать заполнитель(placeholder) `{}`. На его место автоматически подставится нужное значение и не придется использовать дополнительный метод форматирования:
```suggestion
        log.info("requested {} to add", dtoName);
```

Reply:
недосмотрел... стандартная же фича логгера